cpp/src/parquet/column_writer.h

```diff
@@ -91,7 +93,9 @@ class PARQUET_EXPORT PageWriter {
       bool buffered_row_group = false,
       std::shared_ptr<Encryptor> header_encryptor = NULLPTR,
       std::shared_ptr<Encryptor> data_encryptor = NULLPTR,
-      bool page_write_checksum_enabled = false);
+      bool page_write_checksum_enabled = false,
+      ColumnIndexBuilder* column_index_builder = NULLPTR,
```

Comment:
for completeness we should clarify that the builders must outlive the the writer?